torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -824,6 +824,15 @@ class FullyShardedDataParallel(nn.Module):
         same FSDP unit. If enhanced shared parameter support is needed for your
         use case, please ping https://github.com/pytorch/pytorch/issues/77724
 
+    .. note:
+        Attempting to run the forward pass of a submodule that is contained in an
+        FSDP unit is not supported and will result in errors. This is because the
+        submodule's parameters will be sharded, but it itself is not an FSDP instance,
+        so its forward pass will not materialize the full parameters appropriately.
```

Comment:
nit: just a suggestion
```suggestion
        so its forward pass will not all-gather the full parameters appropriately.
```